aleph/logic/graph.py

```diff
@@ -0,0 +1,54 @@
+import logging
+
+from followthemoney import model
+from followthemoney.graph import Graph
+from followthemoney.types import registry
+
+from aleph.index import entities as index
+from aleph.logic.entities import entity_expand_nodes
+
+
+log = logging.getLogger(__name__)
+
+
+class AlephGraph(Graph):
+    def queue(self, id_, proxy=None):
+        if id_ not in self.proxies:
+            if proxy is None:
+                entity = index.get_entity(id_)
+                proxy = model.get_proxy(entity)
+            if proxy is not None:
+                self.proxies[id_] = proxy
```

Comment:
I'm curious if the actual retrieval for this could not be done in `resolve` with `entities_by_id` so that we do only one query instead of N?